crates/oxc_linter/src/rules/unicorn/numeric_separators_style.rs

```diff
@@ -0,0 +1,667 @@
+use lazy_static::lazy_static;
+use oxc_ast::{
+    ast::{BigintLiteral, NumberLiteral},
+    AstKind,
+};
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+use regex::Regex;
+
+use crate::{context::LintContext, fixer::Fix, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint-plugin-unicorn(numeric-separators-style): Enforce the style of numeric separators by correctly grouping digits")]
+#[diagnostic(severity(error), help("Invalid group length in numeric value."))]
```

Comment:
Should be `severity(warning)`, all lints are warnings.

Reply:
It feels like the error message should be "Invalid group length in numeric value." and the help message be something like "Long numbers can become really hard to read, so cutting it into groups of digits, separated with a _, " (from their README)

Reply:
Severity is now `warning`.

I updated the error and help messages, let me know if you'd prefer something else. I tried to find a concise, clear, and descriptive help text.

Error (this is from unicorn's error message):

> Invalid group length in numeric value.

Help:

> Group digits with numeric separators (_) so longer numbers are easier to read.